### ISSUE TITLE
signal.medfilt2d vs ndimage.median_filter

### ISSUE THREAD

**Issue by bcbnz** [scipy/scipy#13509]
In #9680 the speed improvement of using `ndimage.median_filter` over `signal.medfilt` was brought up. The subsequent PR #9685 added a note in the docs suggesting the use of `median_filter` instead. However, it also added the same note to `signal.medfilt2d`, but `medfilt2d` appears to be faster than `median_filter`.

#### Reproducing code example:

Taking the example from #9680 and adapting it to a 2D array of approximately the same number of samples:

```python
import numpy as np
from scipy.signal import medfilt
from scipy.ndimage.filters import median_filter

from timeit import Timer

sig = np.random.random((362, 362))

t_signal = Timer(lambda: medfilt(sig, (9,9)))
t_ndimage = Timer(lambda: median_filter(sig, (9,9), mode='constant'))
print (t_signal.timeit(number=100))
print (t_ndimage.timeit(number=100))

# both give the same result
print(np.allclose(medfilt(sig, (9,9)),median_filter(sig, (9,9), mode='constant')))
```
```
41.575206034000075
9.35733849799999
True
```

So the speed improvement of `median_filter` still holds for 2D filtering. However with `medfilt2d` instead of `medfilt`:

```python
import numpy as np
from scipy.signal import medfilt2d
from scipy.ndimage.filters import median_filter

from timeit import Timer

sig = np.random.random((362, 362))

t_signal = Timer(lambda: medfilt2d(sig, (9,9)))
t_ndimage = Timer(lambda: median_filter(sig, (9,9), mode='constant'))
print (t_signal.timeit(number=100))
print (t_ndimage.timeit(number=100))

# both give the same result
print(np.allclose(medfilt2d(sig, (9,9)),median_filter(sig, (9,9), mode='constant')))
```
```
8.658333002999825
9.530926980999993
True
```

So `medfilt2d` is actually faster -- I see there is specialised code in https://github.com/scipy/scipy/blob/master/scipy/signal/medianfilter.c which implements it so I guess this is not too surprising.

Unless I'm missing something here, I think the note about `median_filter` being faster should be removed from the docstring of `medfilt2d`. And probably `medfilt2d` should at least be mentioned in the "See also" section of the other two functions as a faster version for a specific set of options (2D, zero padded) as a ~10% speed improvement is notable (in my use case with images of ~70,000 by 5,000 pixels the median filter is the slowest part of the processing at the moment).

#### Scipy/Numpy/Python version information:
```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.6.0 1.20.0 sys.version_info(major=3, minor=9, micro=1, releaselevel='final', serial=0)

```

**Comment by grlee77** [comment]
If I recall correctly, `median_filter` is one filter function that scales very well with parallel processing using [dask-image](http://image.dask.org/en/latest/). You can try using [the version there](http://image.dask.org/en/latest/dask_image.ndfilters.html#dask_image.ndfilters.median_filter) and may get an acceleration factor close to the number of CPU cores.

This filter is also available in CuPy (via `cupyx.scipy.ndimage.median_filter`) if you happen to have a compatible GPU.

**Comment by bcbnz** [comment]
Closed by #13530 in commit c7ba5c0b.
